pkg/comp-functions/functions/common/pbp.go

```diff
@@ -0,0 +1,78 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	pdbv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ptr "k8s.io/utils/ptr"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	v1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+func AddPDBSettings(obj client.Object) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+	return func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+
+		log := controllerruntime.LoggerFrom(ctx)
```

Comment:
You can use the logger from `svc`. It will have the right context about the current function.


```suggestion
		log := svc.Log
```